### ISSUE TITLE
Add support for nested lambda expressions in rule conditions

### ISSUE THREAD

**Issue by jonechenug** [NRules/NRules.Language#1]
I had use NRules.Language some little time . It is powerful !
But now I have a problem when using nested objects .Think you have a "Order" object , it has a list to store the detail item.  Now I have a rule  that when any item has some product , the order will had a discount .
The rule look like
```
Name "order rule smaple"
when
   var order= Order(x=>x.Items.Any(i=>i.Product.Name=="some product")); 
then 
    order.discount=1;
```
Unfortunately,it is not work.The erro message say " x.Items has not a member  "Any" " . When I debuging,the Items only has its' index member. So the rule only support look like "x.Items[index].Product.Name=="some product"" .

**Comment by snikolayev** [comment]
You are right, nested lambdas are not yet supported. So this is a valid feature request. It's going to take a little while to add this feature, so meanwhile you can have a workaround - encapsulate the lambda expression behind an extension method on the Order object.

**Comment by harsh-chaturvedi** [comment]
@snikolayev  Any updates on this? Support for nested lambda and execution of linq queries would be really helpful.
